e2e_playwright/st_markdown.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+import streamlit as st
+
+st.markdown("This **markdown** is awesome! :sunglasses:")
+
+st.markdown("This <b>HTML tag</b> is escaped!")
+
+st.markdown("This <b>HTML tag</b> is not escaped!", unsafe_allow_html=True)
+
+st.markdown("[text]")
+
+st.markdown("[link](href)")
+
+st.markdown("[][]")
+
+st.markdown("Inline math with $\KaTeX$")
+
+st.markdown(
+    """
+$$
+ax^2 + bx + c = 0
+$$
+"""
+)
+
+st.markdown("# Some header 1")
+st.markdown("## Some header 2")
+st.markdown("### Some header 3")
+
+st.markdown(
+    """
+| Col1      | Col2        |
+| --------- | ----------- |
+| Some      | Data        |
+"""
+)
+
+with st.container():
+    st.markdown("# some really long header " + " ".join(["lol"] * 10))
+    np.random.seed(0)
+    st.table(np.random.randn(10, 20))
```

Comment:
nit: I'm not 100% sure why what this test was covering in the previous versions, but I think it would be sufficient to actually just use a markdown table or any other markdown block here instead of `st.table`. I assume its just testing that markdown titles don't somehow overlap with the content blocks below. 

Reason is that mixing different elements within e2e tests of other elements always creates a bit of overhead that's better to avoid if it isn't necessary. E.g. if we do a visual update for `st.table`, the markdown test would fail as well since we are testing visual aspects of `st.table` within the markdown e2e test.

Reply:
So, I think I would just remove this `st.table` here and just place a small markdown table here for the test.

Reply:
Done